### ISSUE TITLE
Overloaded definitions of async functions don't seem to be picked up

### ISSUE THREAD

**Issue by thomascobb** [microsoft/pyright#549]
**Describe the bug**
`@overload` decorator on functions defined with the async keyword do not seem to work as expected

**To Reproduce**
Place the following code in an editor window:
```python
from typing import overload

@overload
async def func(x: int) -> int: ...

@overload
async def func(x: str) -> str: ...

async def func(x):
    if isinstance(x, int):
        return 32
    else:
        return "that"

async def test_func():
    x = await func("2")
    assert x == "that"
```
Mouse over `x` and see `(variable) x: Literal[32, 'that']`. If you remove all the async/await keywords you see `(variable) x: str`

**Expected behavior**
Mouse over `x` gives `(variable) x: str`

**VS Code extension or command-line**
VS Code extension version: 1.1.25

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I'll work on a fix.

**Comment by erictraut** [comment]
This will be fixed in the next version of Pyright.

**Comment by thomascobb** [comment]
Thanks for the quick work!

**Comment by erictraut** [comment]
This is now implemented in 1.1.26
